docs/reference/remap.cue

```diff
@@ -65,4 +64,205 @@ remap: {
 			description: "Raised when the provided input cannot be parsed."
 		}
 	}
+
+	// TRL type system
+	types: [TypeName=string]: {
+		#Use: "parameter" | "return"
+
+		description: string
+		use: [#Use, ...#Use]
+	}
+
+	types: {
+		"any": {
+			description: "A stand-in for any type."
+			use: ["parameter"]
+		}
+		"array": {
+			description: "A list of items."
+			use: ["parameter", "return"]
+		}
+		"boolean": {
+			description: "`true` or `false`."
+			use: ["parameter", "return"]
+		}
+		"float": {
+			description: "A 64-bit floating-point number."
+			use: ["parameter", "return"]
+		}
+		"map": {
+			description: """
+				A key-value map in which keys are strings and values can be of any TRL type,
+				including maps.
+				"""
+			use: ["parameter", "return"]
+		}
+		"integer": {
+			description: "A 64-bit integer."
+			use: ["parameter", "return"]
+		}
+		"null": {
+			description: "No value."
+			use: ["return"]
+		}
+		"path": {
+			description: "An event field."
+			use: ["parameter"]
+		}
+		"regex": {
```

Comment:
Also, we should clarify that it's the Rust syntax. Should we include these details or link to this site?

https://docs.rs/regex/1.4.2/regex/#syntax

We've also linked to [this Regex tester](https://rustexp.lpil.uk/) to help users test their expressions.

Reply:
Good call on providing links and yeah, no good reason not to provide examples for all of these.